command/cp.go

```diff
@@ -327,18 +336,21 @@ func (c Copy) Run(ctx context.Context) error {
 		srcurl := object.URL
 		var task parallel.Task
 
-		switch {
-		case srcurl.Type == dsturl.Type: // local->local or remote->remote
-			task = c.prepareCopyTask(ctx, srcurl, dsturl, isBatch)
-		case srcurl.IsRemote(): // remote->local
-			task = c.prepareDownloadTask(ctx, srcurl, dsturl, isBatch)
-		case dsturl.IsRemote(): // local->remote
-			task = c.prepareUploadTask(ctx, srcurl, dsturl, isBatch)
-		default:
-			panic("unexpected src-dst pair")
+		if strutil.CheckAllExclude(c.exclude, object.URL.Path) {
```

Comment:
After changing the function name, could you move the condition before the operation?

For example:

```
if strutil.IsFileExcluded(c.exclude, object.URL.Path) {
	continue
}
```

Reply:
It would be good to make the same change in other commands also.

Reply:
changed all of the commands as you suggested.